daemon/system-info.sh

```diff
@@ -286,6 +286,70 @@ elif [ -r /proc/meminfo ] ; then
         TOTAL_RAM="$((TOTAL_RAM*1024))"
 fi
 
+# -------------------------------------------------------------------------------------------------
+# Detect the total system disk space
+
+DISK_SIZE="unknown"
+DISK_DETECTION="none"
+
+if [ "${KERNEL_NAME}" = "Darwin" ]; then
+        types='hfs'
+
+        if (lsvfs | grep -q apfs) ; then
+            types="${types},apfs"
+        fi
+
+        if (lsvfs | grep -q ufs) ; then
+            types="${types},ufs"
+        fi
+
+        DISK_DETECTION="df"
+        total="$(df -t ${types} -k | tail -n +1 | awk '{s+=$2} END {print s}')"
```

Comment:
At least on my macOS systems; this fails with:

```sh
$ df -t hfs,ufs
df: no file systems processed
```

Reply:
I _suspect_ it _might_ have to do with the fact that I'm using GNU coreutils:

```sh
$ which df
/usr/local/opt/coreutils/libexec/gnubin/df
```

We _should_ account for this too as rare as this _may_ be most devs / sysadmins / etc would setup their macOS environments with Brew and override some of the default BSD utilities.

Reply:
But then again the `/bin/df` utility on macOS that comes installed with macOS / Darwin fails differently again:

```sh
$ /bin/df -t hfs,ufs -k | tail -n +1 | awk '{s+=$2} END {print s}'
df: -k: No such file or directory
```

Reply:
In fact are we sure about the usage the `-k` option?

See for example (_with GNU coreutils_):

```sh
prologic@Jamess-iMac
Tue Jan 28 13:27:20
~
 0 0 0
$ df -t nfs -k | tail -n +1 | awk '{s+=$2} END {print s}'
df: no file systems processed


prologic@Jamess-iMac
Tue Jan 28 13:27:22
~
 1 0 0
$ df -t ufs -k | tail -n +1 | awk '{s+=$2} END {print s}'
df: no file systems processed


prologic@Jamess-iMac
Tue Jan 28 13:27:26
~
 1 0 0
$ df -t apfs -k | tail -n +1 | awk '{s+=$2} END {print s}'
489620265

prologic@Jamess-iMac
Tue Jan 28 13:27:30
~
 0 0 0
$ df -t apfs,ufs,nfs -k | tail -n +1 | awk '{s+=$2} END {print s}'
df: no file systems processed
```

Reply:
It's the `-t` option in the case of GNU coreutils that's causing problems.

Personally, I think the correct approach here is to just always use `/bin/df` on macOS and figure out why it's having issues with `-k`.

I suspect this is a case of it being _super_ picky about option ordering (all the online copies of manpages for the macOS `df` list `-t` last in the options, which I had not noticed before).

Reply:
Looks like you fixed it (_I can't follow the comment trail anymore on this PR_)

```sh
NETDATA_TOTAL_DISK_SIZE=752057774080
NETDATA_DISK_DETECTION=df
```